inc/item_operatingsystem.class.php

```diff
@@ -675,4 +675,12 @@ static function processMassiveActionsForOneItemtype(MassiveAction $ma, CommonDBT
       }
       parent::processMassiveActionsForOneItemtype($ma, $item, $ids);
    }
+
+   public function prepareInputForAdd($input) {
+      $item = getItemForItemtype($input['itemtype']);
+      $item->getFromDB($input['items_id']);
+      $input['entities_id'] = $item->getField('entities_id');
```

Comment:
I don't like ``getField``for anything other than dislay, if it fails, you set NOT_AVAILABLE constant in the $input which contains "N/A" string.